### PR TITLE
BaSP:TG-30: List delete employee page

### PR DESCRIPTION
**Pull request by Ericks19**
Bringed a list of employees from the server to Trackgenix page and add funcionality to delete each one

### CODE REVIEW

File: src/Components/Employees/index.js [hunk 1]
```diff
@@ -1,24 +1,49 @@
-import { useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import styles from './employees.module.css';
+import ListEmployees from './listEmployees';
+import Modal from './modal';
 
 function Employees() {
```
Comment (lmalvasia) [inline src/Components/Employees/index.js:6]:
```suggestion
const Employees = () => {
```

File: src/Components/Employees/index.js [hunk 2]
```diff
@@ -1,24 +1,49 @@
...
-  const [employees, saveEmployees] = useState([]);
+  const [listEmployes, saveEmployees] = useState([]);
+  const [showModal, setShowModal] = useState(false);
+  const [selecterEmployee, setSelecterEmployee] = useState(null);
```
Comment (lmalvasia) [inline src/Components/Employees/index.js:9]:
```suggestion
  const [selecterEmployee, setSelecterEmployee] = useState();
```

File: src/Components/Employees/index.js [hunk 3]
```diff
@@ -1,24 +1,49 @@
...
 
   useEffect(() => {
-    fetch(`https://jsonplaceholder.typicode.com/users`)
+    fetch(`http://localhost:4000/employees`)
```
Comment (lmalvasia) [inline src/Components/Employees/index.js:12]:
```suggestion
    fetch(`${process.env.REACT_APP_API_URL}/employees`)
```

File: src/Components/Employees/index.js [hunk 4]
```diff
@@ -1,24 +1,49 @@
...
+    if (response.ok) {
+      saveEmployees([...listEmployes.filter((employee) => employee._id !== _id)]);
+    }
+  };
+
+  const closeModal = () => {
+    setShowModal(false);
+  };
+  const onComfirm = () => {
+    deleteItem(selecterEmployee);
```
Comment (lmalvasia) [inline src/Components/Employees/index.js:32]:
```suggestion
    deleteItem(selecterEmployee);
    setShowModal(false);
```

File: src/Components/Employees/employee.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+const Employee = ({ employee, setShowModal, setSelecterEmployee }) => {
```
Comment (lmalvasia) [inline src/Components/Employees/employee.js:3]:
```suggestion
const EmployeeRow = ({ employee, setShowModal, setSelecterEmployee }) => {
```
or
```suggestion
const EmployeeItem = ({ employee, setShowModal, setSelecterEmployee }) => {
```

Comment (lmalvasia) [reply src/Components/Employees/employee.js:3]: Rename also the file and put this file inside a folder with the same name.

File: src/Components/Employees/index.js [hunk 6]
```diff
@@ -1,24 +1,49 @@
...
+      method: 'DELETE'
+    });
+    if (response.ok) {
+      saveEmployees([...listEmployes.filter((employee) => employee._id !== _id)]);
+    }
+  };
+
+  const closeModal = () => {
+    setShowModal(false);
+  };
```
Comment (lmalvasia) [inline src/Components/Employees/index.js:30]: New line here to separate the functions

File: src/Components/Employees/listEmployees.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import Employee from './employee';
+
+const ListEmployees = ({ listEmployes, deleteItem, setShowModal, setSelecterEmployee }) => {
```
Comment (lmalvasia) [inline src/Components/Employees/listEmployees.js:4]: Put this component inside his own folder

File: src/Components/Employees/modal.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import styles from './modal.module.css';
+
+function Modal(props) {
```
Comment (lmalvasia) [inline src/Components/Employees/modal.js:3]:
```suggestion
const DeleteConfirmationModal = (props) => {
```

File: src/Components/Employees/modal.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  if (!props.show) {
+    return null;
+  }
+  const onComfirm = () => {
+    props.onComfirm();
+    props.closeModal();
+  };
+  const closeModal = () => {
```
Comment (lmalvasia) [inline src/Components/Employees/modal.js:11]: remove this.

File: src/Components/Employees/modal.js [hunk 10]
```diff
@@ -0,0 +1,26 @@
+import styles from './modal.module.css';
+
+function Modal(props) {
+  if (!props.show) {
+    return null;
+  }
+  const onComfirm = () => {
```
Comment (lmalvasia) [inline src/Components/Employees/modal.js:7]: remove this.

File: src/Components/Employees/modal.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+  };
+  const closeModal = () => {
+    props.closeModal();
+  };
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.modal}>
+        <h3>Modal</h3>
+        <button onClick={closeModal}>Close</button>
```
Comment (lmalvasia) [inline src/Components/Employees/modal.js:19]:
```suggestion
        <button onClick={props.closeModal}>Close</button>
```

File: src/Components/Employees/modal.js [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+        <button onClick={onComfirm}>Confirm</button>
```
Comment (lmalvasia) [inline src/Components/Employees/modal.js:20]:
```suggestion
        <button onClick={props.onConfirm}>Confirm</button>
```

File: src/Components/Employees/index.js [hunk 13]
```diff
@@ -1,24 +1,49 @@
...
+  const onComfirm = () => {
+    deleteItem(selecterEmployee);
+  };
+
   return (
     <section className={styles.container}>
+      <Modal show={showModal} closeModal={closeModal} onComfirm={onComfirm} />
```
Comment (lmalvasia) [inline src/Components/Employees/index.js:37]:
```suggestion
      <Modal show={showModal} closeModal={closeModal} onConfirm={onConfirm} />
```

File: src/Components/Employees/index.js [hunk 14]
```diff
@@ -1,24 +1,49 @@
...
+    });
+    if (response.ok) {
+      saveEmployees([...listEmployes.filter((employee) => employee._id !== _id)]);
+    }
+  };
+
+  const closeModal = () => {
+    setShowModal(false);
+  };
+  const onComfirm = () => {
```
Comment (lmalvasia) [inline src/Components/Employees/index.js:31]:
```suggestion
  const onConfirm = () => {
```

File: src/Components/Employees/employeeRow.js [hunk 15]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+const EmployeeRow = ({ employee, setShowModal, setSelecterEmployee }) => {
```
Comment (lmalvasia) [inline src/Components/Employees/employeeRow.js:3]: Put this file inside a folder.

File: src/Components/Employees/EmployeeRow/employeeRow.js [hunk 16]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+const EmployeeRow = ({ employee, setShowModal, setSelecterEmployee }) => {
```
Comment (GinaSchP) [inline src/Components/Employees/EmployeeRow/employeeRow.js:3]:
```suggestion
const EmployeeRow = ({ employee, setShowModal, setSelectedEmployee }) => {
```

File: src/Components/Employees/EmployeeRow/employeeRow.js [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+  return (
+    <tr className="rows">
+      <td> {employee._id} </td>
+      <td> {employee.name} </td>
+      <td> {employee.email} </td>
+      <td>
+        <button
+          onClick={() => {
+            setSelecterEmployee(employee._id);
```
Comment (GinaSchP) [inline src/Components/Employees/EmployeeRow/employeeRow.js:12]:
```suggestion
            setSelectedEmployee(employee._id);
```

File: src/Components/Employees/ListEmployees/listEmployees.js [hunk 18]
```diff
@@ -0,0 +1,31 @@
...
+          </tr>
+        </thread>
+        <tbody>
+          {listEmployes.map((item) => (
+            <EmployeeRow
+              key={item._id}
+              employee={item}
+              deleteItem={deleteItem}
+              setShowModal={setShowModal}
+              setSelecterEmployee={setSelecterEmployee}
```
Comment (GinaSchP) [inline src/Components/Employees/ListEmployees/listEmployees.js:22]:
```suggestion
              setSelectedEmployee={setSelectedEmployee}
```

File: src/Components/Employees/index.js [hunk 19]
```diff
@@ -1,27 +1,54 @@
-import { useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import styles from './employees.module.css';
+import ListEmployees from './ListEmployees/listEmployees';
+import DeleteConfirmationModal from './Modal/modal';
 
-function Employees() {
-  const [employees, saveEmployees] = useState([]);
+const Employees = () => {
+  const [listEmployes, saveEmployees] = useState([]);
```
Comment (GinaSchP) [inline src/Components/Employees/index.js:7]:
```suggestion
  const [listEmployes, setListEmployees] = useState([]);
```

File: src/Components/Employees/index.js [hunk 20]
```diff
@@ -1,27 +1,54 @@
...
+  const [showModal, setShowModal] = useState(false);
+  const [selecterEmployee, setSelecterEmployee] = useState();
```
Comment (GinaSchP) [inline src/Components/Employees/index.js:9]:
```suggestion
  const [selectedEmployee, setSelectedEmployee] = useState();
```

File: src/Components/Employees/index.js [hunk 21]
```diff
@@ -1,27 +1,54 @@
...
 
   useEffect(() => {
-    fetch(`https://jsonplaceholder.typicode.com/users`)
+    fetch(`${process.env.REACT_APP_API_URL}/employees`)
       .then((response) => response.json())
       .then((response) => {
-        saveEmployees(response);
+        saveEmployees(response.data);
```
Comment (GinaSchP) [inline src/Components/Employees/index.js:15]:
```suggestion
        setListEmployees(response.data);
```

File: src/Components/Employees/index.js [hunk 22]
```diff
@@ -1,27 +1,54 @@
...
       });
   }, []);
 
+  const deleteItem = async (_id) => {
+    const response = await fetch(`${process.env.REACT_APP_API_URL}/employees/${_id}`, {
+      method: 'DELETE'
+    });
+    if (response.ok) {
+      saveEmployees([...listEmployes.filter((employee) => employee._id !== _id)]);
```
Comment (GinaSchP) [inline src/Components/Employees/index.js:24]:
```suggestion
      setListEmployees([...listEmployes.filter((employee) => employee._id !== _id)]);
```

File: src/Components/Employees/index.js [hunk 23]
```diff
@@ -1,27 +1,54 @@
...
+    }
+  };
+
+  const closeModal = () => {
+    setShowModal(false);
+  };
+
+  const onConfirm = () => {
+    deleteItem(selecterEmployee);
```
Comment (GinaSchP) [inline src/Components/Employees/index.js:33]:
```suggestion
    deleteItem(selectedEmployee);
```

**Review by lmalvasia — APPROVED** [review]

**Comment by eyrent** [comment]
### QA Review
- List employees ✅
- Delete employee ✅

**PR APPROVED FOR MERGE**

**Review by lmalvasia — APPROVED** [review]

**Pull request merged**